packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts

```diff
@@ -49,21 +49,22 @@ export default function deepCyclicCopyReplaceable<T>(
 
 function deepCyclicCopyObject<T>(object: T, cycles: WeakMap<any, any>): T {
   const newObject = Object.create(Object.getPrototypeOf(object));
-  const descriptors = Object.getOwnPropertyDescriptors(object);
+  const descriptors: {
+    [x: string]: PropertyDescriptor;
+  } = Object.getOwnPropertyDescriptors(object);
 
   cycles.set(object, newObject);
 
   Object.keys(descriptors).forEach(key => {
-    const descriptor = descriptors[key];
-    if (typeof descriptor.value !== 'undefined') {
-      descriptor.value = deepCyclicCopyReplaceable(descriptor.value, cycles);
-    }
-
-    if (!('set' in descriptor)) {
-      descriptor.writable = true;
-    }
-
-    descriptor.configurable = true;
+    descriptors[key] = {
```

Comment:
Can the enumerable check be here instead?

```js
if (descriptors[key].enumerable) {
  descriptors[key] = ...
} else {
  delete descriptors[key];
}
```

Reply:
in that case we lose `Symbol` keys. Not sure why...

Reply:
At least if I understood your comment correcly


```diff
diff --git i/packages/jest-matcher-utils/src/Replaceable.ts w/packages/jest-matcher-utils/src/Replaceable.ts
index fc8c7343c..dd9e04f32 100644
--- i/packages/jest-matcher-utils/src/Replaceable.ts
+++ w/packages/jest-matcher-utils/src/Replaceable.ts
@@ -35,12 +35,6 @@ export default class Replaceable {
       Object.entries(this.object).forEach(([key, value]) => {
         cb(value, key, this.object);
       });
-      Object.getOwnPropertySymbols(this.object).forEach(key => {
-        const descriptor = Object.getOwnPropertyDescriptor(this.object, key);
-        if (descriptor?.enumerable) {
-          cb(this.object[key], key, this.object);
-        }
-      });
     } else {
       this.object.forEach(cb);
     }
diff --git i/packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts w/packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts
index ad98131dd..67271d5d8 100644
--- i/packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts
+++ w/packages/jest-matcher-utils/src/deepCyclicCopyReplaceable.ts
@@ -56,15 +56,19 @@ function deepCyclicCopyObject<T>(object: T, cycles: WeakMap<any, any>): T {
   cycles.set(object, newObject);
 
   Object.keys(descriptors).forEach(key => {
-    descriptors[key] = {
-      configurable: true,
-      enumerable: true,
-      value: deepCyclicCopyReplaceable(
-        (object as Record<string, unknown>)[key],
-        cycles,
-      ),
-      writable: true,
-    };
+    if (descriptors[key].enumerable) {
+      descriptors[key] = descriptors[key] = {
+        configurable: true,
+        enumerable: true,
+        value: deepCyclicCopyReplaceable(
+          (object as Record<string, unknown>)[key],
+          cycles,
+        ),
+        writable: true,
+      };
+    } else {
+      delete descriptors[key];
+    }
   });
 
   return Object.defineProperties(newObject, descriptors);
```

Reply:
You'd still need the `Object.getOwnPropertySymbols` call but you can lose the `if (descriptor?.enumerable) {` check because we would've not copied those over in the first place.

Reply:
right! done 🙂